tests/test_gcm.py

```diff
@@ -371,6 +378,29 @@ def test_misformed_default_payload_rejected(self) -> None:
         self.assertEqual(resp, {"rejected": ["badpayload"]})
         self.assertEqual(gcm.num_requests, 0)
 
+    def test_empty_payload_and_no_data_rejected(self) -> None:
+        """
+        Tests that an empty default_payload with no other data to send is rejected.
+        This results in a notification with no content, so we reject it.
+        """
+        gcm = self.get_test_pushkin("com.example.gcm")
+        gcm.preload_with_response(
+            200, {"results": [{"message_id": "msg42", "registration_id": "emptypayload"}]}
+        )
+
+        resp = self._request(
+            {
+                "notification": {
+                    "id": "$3957tyerfgewrf384",
+                    "prio": "high",
+                    "devices": [DEVICE_EXAMPLE_WITH_EMPTY_DEFAULT_PAYLOAD],
+                }
+            }
+        )
+
+        self.assertEqual(resp, {"rejected": ["emptypayload"]})
```

Comment:
The spec [instructs](https://spec.matrix.org/v1.16/push-gateway-api/#post_matrixpushv1notify) homeservers to delete the pusher associated with any rejected requests(!), as they should only be rejected if the pushkeys are invalid.

So rather than rejecting them, we should instead silently ignore them.